csharp/src/Apache.Arrow.Flight/Action.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Google.Protobuf;
+
+namespace Apache.Arrow.Flight
+{
+    public class Action
```

Comment:
This is kind of an unfortunate name since it conflicts with `System.Action`. Any time someone tries using this class (or System.Action) and they have both namespaces in their `usings`, they will get:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0104	'Action' is an ambiguous reference between 'Apache.Arrow.Flight.Action' and 'System.Action'	ArrowFlightTest	F:\LinuxShare\ArrowFlight\Program.cs	15	Active
```

Reply:
@eerhardt I think I will actually prefix all names by Flight if that sounds ok to you as well. A lot of the names are quite generic such as Action, Location, Result, Criteria. There mightbe conflicts with other libraries for the other names as well. So I were thinking that having the same prefix gives it some kind of uniformity.

Reply:
That makes sense to me.